### PR TITLE
[Core][AnalysisStage] Using Check

### PR DESCRIPTION
**Pull request by philbucher**
This PR introduces the usage of `Check` in the `AnalysisStage`
So far this was done manually in every solver (or at least in some of them)

Now it is more unified and also prepares for checking the Processes => #3002 

I also checked the solvers in FSI and ContactStructuralMechanics

### CODE REVIEW

**Comment by philbucher** [comment]
ping

**Comment by philbucher** [comment]
ping @KratosMultiphysics/technical-committee

**Comment by philbucher** [comment]
@KratosMultiphysics/technical-committee a very good example of how this can be used is in #3274

**Comment by philbucher** [comment]
ping

File: kratos/python_scripts/analysis_stage.py [hunk 1]
```diff
@@ -78,6 +78,7 @@ def Initialize(self):
             process.ExecuteInitialize()
 
         self._GetSolver().Initialize()
+        self.Check()
```
Comment (RiccardoRossi) [inline kratos/python_scripts/analysis_stage.py:81]: I appreciate that there may be issues with this...however shouldn't we do it BEFORE the initialize?

to make it clear, I DO SEE that you are not changing the current behaviour :-)

**Review by RiccardoRossi — DISMISSED** [review]

File: kratos/python_scripts/analysis_stage.py (hunk 1 above)
Comment (philbucher) [reply kratos/python_scripts/analysis_stage.py:81]: Hm I honestly don't know what would be better ... As you said I just unified the current behavior

Either way, if we decide to change this then we will have to do it only in one place if this is merged :)
Therefore I suggest that I merge this and we can discuss when to call `Check` separately
**I will merge on Friday unless someone objects**

If you want I can also open an issue regarding when to call `Check`

Thanks btw for discussing this in the committee

**Comment by philbucher** [comment]
can someone reapprove pls?
I had to solve a conflict

**Review by loumalouomega — APPROVED** [review]

**Pull request merged**